File: go/map/map.go
package selfmap

// 算了算了，还是在没搞清楚 go 的泛型前不折腾泛型了
type Map interface {
	Add(key string, value int)
	Remove(key string) (int, bool)
	Contains(key string) bool
	Get(key string) (int, bool)
	Set(key string, value int) error
	GetSize() int
	IsEmpty() bool
}

File: js2/map.ts
export interface Map<K, V> {
    add(k: K, v: V): void;
    remove(k: K): V | null;
    contains(k: K): boolean;
    get(k: K): V | null;
    set(k: K, v: V): void;
    getSize(): number;
    isEmpty(): boolean;
}
